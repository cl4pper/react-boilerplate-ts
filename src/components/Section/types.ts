export interface SectionProps {
    title: String,
} 