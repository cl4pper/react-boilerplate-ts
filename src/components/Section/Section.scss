.Section {
    &__text {
        &--hidden {
            visibility: hidden;
        }
    }
}