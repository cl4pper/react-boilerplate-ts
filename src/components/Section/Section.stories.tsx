import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { Section } from './Section';
import * as Types from './types';

const stories = storiesOf('Section', module);
stories.addDecorator(withKnobs);

const props: Types.SectionProps = {
    title: 'Section title comes here'
}

stories.add('Default', () => (
    <Section title={text('title', props.title, null)} />
));