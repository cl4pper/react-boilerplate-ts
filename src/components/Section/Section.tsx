import React from 'react';

// STYLE
import './Section.scss';

// TYPES
import * as Types from './types';

const Section = (props: Types.SectionProps) => {
    const { title } = props;

    return (
        <section className="Section">
            <h1 className="Section__title">{title}</h1>
        </section>
    )
}

export {
    Section
}