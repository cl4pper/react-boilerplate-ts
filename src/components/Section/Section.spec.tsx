import React from 'react';
import { shallow } from 'enzyme';

import { Section } from './Section';
import * as Types from './types';

const titleProp: string = 'Test title';
const props: Types.SectionProps = {
    title: titleProp,
};

describe('Section component', () => {
    const wrapper = shallow(<Section {...props} />);

    it('should has class Section', () => {
        expect(wrapper.is('.Section')).toBe(true);
    });

    it('should return a title passed by prop title', () => {
        expect(wrapper.text()).toBe(titleProp);
    });
})
