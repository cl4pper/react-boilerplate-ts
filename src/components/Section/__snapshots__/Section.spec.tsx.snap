// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section 1`] = `ShallowWrapper {}`;
